app/models/user_export.rb

```diff
@@ -3,10 +3,8 @@ class UserExport < ActiveRecord::Base
 
   def self.remove_old_exports
     UserExport.where('created_at < ?', 2.days.ago).find_each do |user_export|
-      file_name = "#{user_export.file_name}-#{user_export.id}.csv.gz"
-      file_path = "#{UserExport.base_directory}/#{file_name}"
-
-      File.delete(file_path) if File.exist?(file_path)
+      upload = Upload.find_by(id: user_export.upload_id)
+      upload.destroy! if upload.present?
       user_export.destroy!
```

Comment:
Perhaps we can set a `belongs_to :upload, dependent: :destroy`  on `UserExport`?

Reply:
Good idea! 👍